README.md
# sdv

[![Commitizen friendly](https://img.shields.io/badge/commitizen-friendly-brightgreen.svg)](http://commitizen.github.io/cz-cli/)

A nice little utility helper for Stardew Valley

https://sdv.msroed.now.sh

## Getting Started

### Install dependencies

`yarn`

### Run the development server

`yarn dev`

### Build

`yarn build`

### Run full production server locally

`yarn start`

### Style and formatting

`yarn lint` will run `ESLint` on all the staged files.

## Deployment

All commits will be deployed to its own URL. Commiting on master will deploy to production.

## Contributing

Feel free to send a pull request. The repository is commitizen-friendly.

Running `yarn commit`

will guide you in creating a nice commit -- this is a commitizen friendly repo.

## Acknowledgments

ConcernedApe for being absolutely amazing


CHANGELOG.md
# [1.13.0](https://github.com/martolini/sdv/compare/v1.12.3...v1.13.0) (2020-10-04)


### Features

* Force release ([0fea55f](https://github.com/martolini/sdv/commit/0fea55f0d91c4403d75174dfc7d3da1bdb2e648e))

## [1.12.3](https://github.com/martolini/sdv/compare/v1.12.2...v1.12.3) (2020-10-04)


### Bug Fixes

* 🐛 Add error reporting ([7815c8e](https://github.com/martolini/sdv/commit/7815c8efe9e66795b3122fb87eb1cc0ee0f94d79))


### Reverts

* Revert "fix: 🐛 Add error reporting" (#47) ([0b711b7](https://github.com/martolini/sdv/commit/0b711b7a6a0e447857b6e40d0ead5e8fea3364a2)), closes [#47](https://github.com/martolini/sdv/issues/47)

## [1.12.2](https://github.com/martolini/sdv/compare/v1.12.1...v1.12.2) (2020-01-19)


### Bug Fixes

* 🐛 Disable press f to search ([76da8ce](https://github.com/martolini/sdv/commit/76da8ce01b76fb8f02f7e0e17eb11def5e5633c7))
* 🐛 Fix resilience in uploads and parser ([c86e298](https://github.com/martolini/sdv/commit/c86e29811b5ecd1bbb8fb5836d0abaa1d999aff2))

## [1.12.1](https://github.com/martolini/sdv/compare/v1.12.0...v1.12.1) (2020-01-19)


### Bug Fixes

* 🐛 Safeguard location.item in findHarvestOnFarm ([28a65f3](https://github.com/martolini/sdv/commit/28a65f3388dcf46ec90a6e8c4b0014aaed60c745))
* 🐛 Upload the xml before trying to parse it ([731ad56](https://github.com/martolini/sdv/commit/731ad56d000b5c77a999839643a5866a2b520952))

# [1.12.0](https://github.com/martolini/sdv/compare/v1.11.2...v1.12.0) (2020-01-19)


### Features

* 🎸 Removed mixpanel in favor of segment ([dfa3a03](https://github.com/martolini/sdv/commit/dfa3a03c50996d6313dba88d82edcbf4e15f7239))

## [1.11.2](https://github.com/martolini/sdv/compare/v1.11.1...v1.11.2) (2020-01-19)


### Reverts

* Revert "fix: 🐛 Let now github app deploy to production prematurely" ([84bfd76](https://github.com/martolini/sdv/commit/84bfd76b5825c232fe32ecbb1aa9b3d0e414e9a2))

## [1.11.1](https://github.com/martolini/sdv/compare/v1.11.0...v1.11.1) (2020-01-19)


### Bug Fixes

* 🐛 Let now github app deploy to production prematurely ([7604efe](https://github.com/martolini/sdv/commit/7604efe8d7590ae11a2467f9a528459da351a2ed))

# [1.11.0](https://github.com/martolini/sdv/compare/v1.10.0...v1.11.0) (2020-01-19)


### Features

* 🎸 Add meta tags ([#38](https://github.com/martolini/sdv/issues/38)) ([1af80e8](https://github.com/martolini/sdv/commit/1af80e8348b84676812aaa339ea10bb3e894b9b5))

# [1.10.0](https://github.com/martolini/sdv/compare/v1.9.1...v1.10.0) (2020-01-19)


### Features

* 🎸 Add mixpanel ([#37](https://github.com/martolini/sdv/issues/37)) ([21aeb25](https://github.com/martolini/sdv/commit/21aeb2508deecd30049c30993c23ae7808af8f5f))

## [1.9.1](https://github.com/martolini/sdv/compare/v1.9.0...v1.9.1) (2020-01-19)


### Bug Fixes

* 🐛 Fix build workflows ([471c9c7](https://github.com/martolini/sdv/commit/471c9c7c861660d486e51fd562030f72fd8f42b1))

# [1.9.0](https://github.com/martolini/sdv/compare/v1.8.1...v1.9.0) (2020-01-19)


### Bug Fixes

* 🐛 Add target=_blank to links in Footer ([#34](https://github.com/martolini/sdv/issues/34)) ([5fe6c3d](https://github.com/martolini/sdv/commit/5fe6c3de54b881dff1b065af22ac7b6c393f611b))


### Features

* 🎸 Add welcome msg for first time users and example farm ([#35](https://github.com/martolini/sdv/issues/35)) ([bd4fd7a](https://github.com/martolini/sdv/commit/bd4fd7ae99ec37006422a0fb2022eeb8c9622a04))

## [1.8.1](https://github.com/martolini/sdv/compare/v1.8.0...v1.8.1) (2020-01-18)


### Bug Fixes

* 🐛 Runtime error in Antd menu ([#32](https://github.com/martolini/sdv/issues/32)) ([e432336](https://github.com/martolini/sdv/commit/e432336b256ff18e0c548c5c2412865a60235cdb))

# [1.8.0](https://github.com/martolini/sdv/compare/v1.7.1...v1.8.0) (2020-01-18)


### Features

* 🎸 Add feedback to menubar ([#29](https://github.com/martolini/sdv/issues/29)) ([6a4f461](https://github.com/martolini/sdv/commit/6a4f461ec34b98525be13e01a8d00ccef7173cfe))
* 🎸 Change /user to /farmer ([#28](https://github.com/martolini/sdv/issues/28)) ([892fd9b](https://github.com/martolini/sdv/commit/892fd9b4e81387a0146889de56d12dfa826fc694))

## [1.7.1](https://github.com/martolini/sdv/compare/v1.7.0...v1.7.1) (2020-01-18)


### Bug Fixes

* Bump package.json version on release ([90cc6cf](https://github.com/martolini/sdv/commit/90cc6cfaff02936dd69cc91cffecc0cc3b84e167))

# [1.7.0](https://github.com/martolini/sdv/compare/v1.6.0...v1.7.0) (2020-01-18)


### Features

* Add changelog and link to it from application ([#25](https://github.com/martolini/sdv/issues/25)) ([b282e46](https://github.com/martolini/sdv/commit/b282e46126c8b5e26faa7d086f55e6de912fde34))


pages/user.js
import React from 'react';
import { useStoreState } from 'easy-peasy';
import { Progress, Row, Col, Empty, Tooltip } from 'antd';

const SKILL_TABLE = {
  0: 'Farming',
  1: 'Fishing',
  2: 'Foraging',
  3: 'Mining',
  4: 'Combat',
};

const EXP_TABLE = {
  0: 0,
  1: 100,
  2: 380,
  3: 770,
  4: 1300,
  5: 2150,
  6: 3300,
  7: 4800,
  8: 6900,
  9: 10000,
  10: 15000,
};

const PROFESSIONS_TABLE = {
  Farming: {
    0: { name: 'Rancher', description: 'Animal products worth 20% more.' },
    1: { name: 'Tiller', description: 'Crops worth 10% more.' },
    2: {
      name: 'Coopmaster',
      description:
        'Befriend coop animals quicker.\nIncubation time cut in half.',
    },
    3: {
      name: 'Shepherd',
      description: 'Befriend barn animals quicker.\nSheep produce wool faster.',
    },
    4: {
      name: 'Artisan',
      description: 'Artisan goods (wine, cheese, oil, etc.) worth 40% more.',
    },
    5: { name: 'Agriculturist', description: 'All crops grow 10% faster.' },
  },
  Fishing: {
    6: { name: 'Fisher', description: 'Fish worth 25% more.' },
    7: {
      name: 'Trapper',
      description: 'Resources required to craft crab pots reduced.',
    },
    8: { name: 'Angler', description: 'Fish worth 50% more.' },
    9: { name: 'Pirate', description: 'Chance to find treasure doubled.' },
    10: {
      name: 'Mariner',
      description: 'Crab pots no longer produce junk items.',
    },
    11: {
      name: 'Luremaster',
      description: 'Crab pots no longer require bait.',
    },
  },
  Foraging: {
    12: { name: 'Forester', description: 'Trees drop 25% more wood.' },
    13: {
      name: 'Gatherer',
      description: 'Chance for double harvest of foraged items.',
    },
    14: {
      name: 'Lumberjack',
      description: 'All trees have a chance to drop hardwood.',
    },
    15: { name: 'Tapper', description: 'Syrups worth 25% more.' },
    16: {
      name: 'Botanist',
      description: 'Foraged items are always highest quality.',
    },
    17: {
      name: 'Tracker',
      description: 'Location of forageable items revealed.',
    },
  },
  Mining: {
    18: { name: 'Miner', description: '+1 ore per vein.' },
    19: {
      name: 'Geologist',
      description: 'Chance for gems to appear in pairs.',
    },
    20: { name: 'Blacksmith', description: 'Metal bars worth 50% more.' },
    21: { name: 'Prospector', description: 'Chance to find coal doubled.' },
    22: { name: 'Excavator', description: 'Chance to find geodes doubled.' },
    23: { name: 'Gemologist', description: 'Gems worth 30% more.' },
  },
  Combat: {
    24: { name: 'Fighter', description: 'All attacks deal 10% more damage.' },
    25: {
      name: 'Scout',
      description: 'Critical strike chance increased by 50%.',
    },
    26: { name: 'Brute', description: 'Deal 15% more damage.' },
    27: { name: 'Defender', description: '+25 HP.' },
    28: {
      name: 'Acrobat',
      description: 'Cooldown on special moves cut in half.',
    },
    29: { name: 'Desperado', description: 'Critical strikes are deadly.' },
  },
};

export default function PlayerView(props) {
  const players = useStoreState(state => state.players);
  if (Object.keys(players).length === 0) {
    return <Empty />;
  }
  const renderObject = Object.keys(players).map(playerName => {
    const player = players[playerName];
    const {
      experiencePoints: { int: experiencePoints },
    } = player;
    const levels = Object.keys(SKILL_TABLE).map(skillKey => {
      const skillName = SKILL_TABLE[skillKey];
      const stateKey = `${skillName.toLowerCase()}Level`;
      const level = player[stateKey];
      const exp = EXP_TABLE[level];
      const nextExp = EXP_TABLE[level + 1];
      const percentage =
        level === 10
          ? 100
          : ((experiencePoints[skillKey] - exp) / (nextExp - exp)) * 100;
      const chosenProfessions = player.professions.int
        .filter(prof => !!PROFESSIONS_TABLE[skillName][prof])
        .map((prof, idx) => (
          <Tooltip
            key={idx}
            placement="right"
            title={PROFESSIONS_TABLE[skillName][prof].description}
          >
            <div style={{ color: 'darkgray' }}>
              #{idx + 1}: {PROFESSIONS_TABLE[skillName][prof].name}
            </div>
          </Tooltip>
        ));
      return (
        <Col span={4} key={skillKey}>
          <div style={{ display: 'inline-block' }}>
            {skillName}: {level}
            {chosenProfessions}
          </div>
          <Progress
            type="circle"
            key={skillKey}
            percent={Math.round(percentage)}
            style={{ paddingLeft: 10 }}
          />
        </Col>
      );
    });
    return (
      <div key={playerName}>
        <h1>{playerName}</h1>
        <Row>{levels}</Row>
      </div>
    );
  });
  return <div>{renderObject}</div>;
}
